prov/sockets/src/sock_ep_msg.c

```diff
@@ -693,8 +694,10 @@ static void *sock_msg_ep_listener_thread(void *data)
 	poll_fds[0].events = poll_fds[1].events = POLLIN;
 
 	while (*((volatile int*) &ep->attr->cm.do_listen)) {
+		fastlock_acquire(&ep->attr->cm.lock);
 		timeout = dlist_empty(&ep->attr->cm.msg_list) ? -1 :
 					SOCK_CM_COMM_TIMEOUT;
+		fastlock_release(&ep->attr->cm.lock);
```

Comment:
On Jun 23, 2016, at 8:45 PM, Shantonu Hossain <notifications@github.com> wrote:
> 
> Sure, Jithin already has a PR that refactors the connection management for MSG ep. We can revisit this issue when he is back.

Ah -- I did not see https://github.com/ofiwg/libfabric/pull/2119.  Maybe that will fix these issues...